.github/workflows/code-checks.yml

```diff
@@ -96,8 +87,22 @@ jobs:
       run: ci/code_checks.sh docstrings
       if: ${{ steps.build.outcome == 'success' }}
 
-    - name: Run typing validation
-      run: ci/code_checks.sh typing
+    - name: Use existing environment for type checking
+      run: |
+        echo $PATH >> $GITHUB_PATH
+        echo "PYTHONHOME=$PYTHONHOME" >> $GITHUB_ENV
+        echo "PYTHONPATH=$PYTHONPATH" >> $GITHUB_ENV
+
+    - name: Mypy
+      uses: pre-commit/action@v2.0.3
+      with:
+        extra_args: --hook-stage manual --all-files mypy
```

Comment:
Is it possible to both run mypy and pyright in this step and have this be a `- name: Typing` check?

Reply:
As far as I know, that is possible if mypy&pyright are in the same stage (that is the case) AND if no other hook is in this stage.

I'm not sure how people use per-commit, could reserve `manual` for mypy&pyright.

Reply:
That sounds good to me (for now). We could reserve manual for more expensive code checks like these

Reply:
Somehow two of the hooks are not affected by `default_stages`, so I explicitly set their stages.